packages/next/build/webpack/plugins/flight-types-plugin.ts

```diff
@@ -35,8 +35,8 @@ interface LayoutProps {
   params: any
 }
 
-type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>
-type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>
+export type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>
+export type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>
```

Comment:
Instead of exporting can't you declare these types based on `options.type === 'layout'`?

Reply:
These types _might_ be useful for lib authors so I think exporting is fine, but we can always iterate on it in the future. `app/` is not yet stable, but thanks for the feedback!